options.go

```diff
@@ -51,6 +51,14 @@ func WithLogger(msg *log.Logger) Option {
 	}
 }
 
+// WithDialerMaxRetries configures the maximum number of retries
+// when dialing an endpoint (-1 means infinite retries).
+func WithDialerMaxRetries(maxRetries int) Option {
```

Comment:
perhaps add an example or a test (or modify an example/test) to use that new public API?

Reply:
I added a test